### PR TITLE
Feature/u3 improvements eventlog

### PR DESCRIPTION
**Pull request by harrwiss**
**Security Log**
- added tcp only filter checkbox and remote host name column 
- added DnsResolver to resolve remote ip's to hostname using dns asynchronous
- improved filter lookup to include wfp state which includes filters from other filter providers besides windows firewall (e.g third party firewall)

**Options**
- added Enable DNS host name resolver setting for Security Log
- added Enable successful connection events setting for Security Log (auditpol /success:enable)

**TestProject**
- added additional tests for NetshHelper and DnsResolver
- cleanup namespace

**Various**
- remove old ip database (IPDatabase.gz)
- cleanup and fix some warnings

### CODE REVIEW

**Comment by AtlasHackert** [comment]
In LogEntryViewModel.cs, what is this "Harrwiss.Common.Network.Helper" you are using?

**Comment by harrwiss** [comment]
Sorry, I overlooked the message in the inbox. The namespace is of course used because I created the new classes. Why is this a problem? Thx.

**Comment by AtlasHackert** [comment]
Well, I'm not going to be the one that pulls in new foreign namespaces.
@Wokhan: What's your opinion on this?

**Comment by wokhan** [comment]
Hi,
First thank you @harrwiss for your involvement, and thank you (again) @atlashackert for your usual help :-) 
Regarding the namespace matter: if you contribute to a project, you have to follow its "standards", including existing root namespace. Imagine how messy it would be if all contributors were to add their own? (well, that's only 2 for WFN but you get my point 😁). Of course you can add credit in comments or documentation, if you want to be recognized for your work (makes sense).
Sorry about that, but even if WFN isn't really cleanly coded (it's a 10 or 11 years old project and I should probably refactor it a bit again... especially because of all improvements in the .Net area those last years), we have to set some base rules 😊
Thanks

**Comment by wokhan** [comment]
Oh and the initial root namespace (wokhan) is indeed also how I appear on GitHub, but only because my nickname (khan) was taken... And then wokhan (for the github organization) was taken by me (haha) so I created wokhansoft to also match twitter (which I don't use that much...). Yeah that's a bit hard to follow. If we want to be consistent maybe we should move to WokhanSoft as the main root but I think it's not worth it (and I'll add other projects / libs which share the same "wokhan" moniker soon).
Maybe renaming the organization (and my user) would be actually easier 😁

**Comment by harrwiss** [comment]
Hi both, thanks for clarifying the matter and yes there need to be some rules though its a pain sometimes. Any objections about the namespace "TestProject" used in the TestProject or can we keep that at least? 
Thx.

**Comment by wokhan** [comment]
Hi, test projects should be named after their "target", i.e. Console.Tests.csproj in a Console.Tests folder.
For usual code rules, I tend to follow MS rules (through the Roslyn linter) but as far as I remember I wasn’t so open / aware when I was working on WFN 😁 so it might differ... Anyway, I think it's better to follow Roslyn's and thus MS rules.

Note: I think I'll update the project to .net core 3 as I did on other private WPF projects, it will allow better performance and would motivate me to do some refactoring. For tests we could then use xUnit (didn’t check what you use now), we can already btw but would be better to upgrade first. Users who don't want to rely on external .net fwk updates would then be able to use a standalone version, amongst other improvements.. 

Thanks !

**Comment by harrwiss** [comment]
Hi wokhan, since I have not much experience with .NET development it's more a best guess for me to decide what would be appropriate - my vote would be to use the java formatter rules though ;)  

> For usual code rules, I tend to follow MS rules (through the Roslyn linter) 

I think it should be possible to set the VS rules applied in the WFN project properties - think that would be helpful and would be something for the project master to decide and setup :)

> Test projects should be named after their "target", i.e. Console.Tests.csproj in a Console.Tests folder

Makes sense to create one test project per project. The namespace would then be the same as the one in the class file under test I guess. For me it's just using the VS Community edition there is not much help in refactoring :( so I'll do it when I'm in the mood ;)

Currently I use namespace "TestProject.ConsoleTests" for instance, because "TestProject.Console" has some conflict with System.Console when I want to use Console.writeLine :(

> I think I'll update the project to .net core 3 as I did on other private WPF projects

👍 

> For tests we could then use xUnit (didn’t check what you use now)

Think I took "Unit Test Project (MSTest)" from the selection but don't know hot to check now 😬 If xUnit is the choice then its ok for me.

Cheers.

**Comment by harrwiss** [comment]
Taking it back  now - will provide another pull request asap.
